sdks/python/tests/e2e/test_tracing.py

```diff
@@ -655,3 +671,266 @@ def f():
+        assert attachment.link.startswith(url_override)
+
+
+def test_tracked_function__update_current_trace__with_attachments(
+    opik_client, data_file
+):
+    file_name = os.path.basename(data_file.name)
+    # Setup
+    ID_STORAGE = {}
+    THREAD_ID = id_helpers.generate_id()
+
+    @opik.track
+    def f():
+        opik_context.update_current_trace(
+            name="trace-name",
+            input={"trace-input": "trace-input-value"},
+            output={"trace-output": "trace-output-value"},
+            metadata={"trace-metadata-key": "trace-metadata-value"},
+            thread_id=THREAD_ID,
+            attachments=[
+                Attachment(
+                    data=data_file.name,
+                    file_name=file_name,
+                    content_type="application/octet-stream",
+                ),
+            ],
+        )
+        ID_STORAGE["f_trace-id"] = opik_context.get_current_trace_data().id
+
+    # Call
+    f()
+    opik.flush_tracker()
+
+    # check that the attachment was uploaded
+    trace_id = ID_STORAGE["f_trace-id"]
+
+    url_override = opik_client._config.url_override
+    url_override_path = base64.b64encode(url_override.encode("utf-8")).decode("utf-8")
+    trace = opik_client.get_trace_content(id=trace_id)
+    attachments = opik_client._rest_client.attachments.attachment_list(
+        project_id=trace.project_id,
+        entity_type="trace",
+        entity_id=trace_id,
+        path=url_override_path,
+    )
+
+    assert len(attachments.content) == 1
+    attachment = attachments.content[0]
+    assert attachment.file_name == file_name
+    assert attachment.mime_type == "application/octet-stream"
+    assert attachment.file_size == FILE_SIZE
+    assert attachment.link.startswith(url_override)
+
+
+def test_opik_span__attachments(opik_client, data_file):
+    trace_id = helpers.generate_id()
+    file_name = os.path.basename(data_file.name)
+    names = [file_name + "_first", file_name + "_second"]
+
+    # Send a trace that matches the input filter
+    opik_client.trace(
+        id=trace_id,
+        name="trace-name",
+        input={"input": "Some random input"},
+        output={"output": "trace-output"},
+        project_name=OPIK_E2E_TESTS_PROJECT_NAME,
+    )
+    span = opik_client.span(
+        trace_id=trace_id,
+        name="span-name",
+        input={"input": "Some random input 2"},
+        output={"output": "span-output"},
+        attachments=[
+            Attachment(
+                data=data_file.name,
+                file_name=names[0],
+                content_type="application/octet-stream",
+            ),
+            Attachment(
+                data=data_file.name,
+                file_name=names[1],
+                content_type="application/octet-stream",
+            ),
+        ],
+    )
+
+    opik_client.flush()
+
+    # check that the attachment was uploaded
+    url_override = opik_client._config.url_override
```

Comment:
Let's create `verify_attachment` or `verify_attachments` and encapsulate those api calls and assertions into it, similarly to verify_span and verify_trace (I recommend looking at the structure of those verifiers, they also encapsulate the logic of trying to get the data multiple times if the first ones were not successful)